sklearn/decomposition/tests/test_online_lda.py

```diff
@@ -0,0 +1,292 @@
+
+    correct_idx_grps = [(0, 1, 2), (3, 4, 5), (6, 7, 8)]
+    for c in lda.components_:
+        top_idx = set(c.argsort()[-3:][::-1])
+        assert_true(tuple(sorted(top_idx)) in correct_idx_grps)
+
+
+def test_lda_perplexity():
+    """
+    Test LDA perplexity for batch training
+    perplexity should be lower after each iteration
+    """
+    n_topics, doc_topic_prior, topic_word_prior, X = _build_sparse_mtx()
+    for method in ('online', 'batch'):
+        lda_1 = LatentDirichletAllocation(n_topics=n_topics, doc_topic_prior=doc_topic_prior,
+                                          topic_word_prior=topic_word_prior, max_iter=1,
+                                          learning_method=method, total_samples=100, random_state=0)
+        lda_2 = LatentDirichletAllocation(n_topics=n_topics, doc_topic_prior=doc_topic_prior,
+                                          topic_word_prior=topic_word_prior, max_iter=10,
+                                          learning_method=method, total_samples=100, random_state=0)
+        distr_1 = lda_1.fit_transform(X)
+        perp_1 = lda_1.perplexity(X, distr_1, sub_sampling=False)
+
+        distr_2 = lda_2.fit_transform(X)
+        perp_2 = lda_2.perplexity(X, distr_2, sub_sampling=False)
+        assert_greater_equal(perp_1, perp_2)
+
+
+def test_lda_score():
+    """
+    Test LDA score for batch training
+    score should be higher after each iteration
+    """
+    n_topics, doc_topic_prior, topic_word_prior, X = _build_sparse_mtx()
+    for method in ('online', 'batch'):
+        lda_1 = LatentDirichletAllocation(n_topics=n_topics, doc_topic_prior=doc_topic_prior,
+                                          topic_word_prior=topic_word_prior, max_iter=1,
+                                          learning_method=method, total_samples=100, random_state=0)
+        lda_2 = LatentDirichletAllocation(n_topics=n_topics, doc_topic_prior=doc_topic_prior,
+                                          topic_word_prior=topic_word_prior, max_iter=10,
+                                          learning_method=method, total_samples=100, random_state=0)
+        lda_1.fit_transform(X)
+        score_1 = lda_1.score(X)
+
+        lda_2.fit_transform(X)
+        score_2 = lda_2.score(X)
+        assert_greater_equal(score_2, score_1)
+
+
+def test_lda_score_perplexity():
+    """
+    Test the relationship between LDA score and perplexity
+    """
+    n_topics, doc_topic_prior, topic_word_prior, X = _build_sparse_mtx()
+    lda = LatentDirichletAllocation(n_topics=n_topics, doc_topic_prior=doc_topic_prior,
+                                    topic_word_prior=topic_word_prior, max_iter=10,
+                                    random_state=0)
+    distr = lda.fit_transform(X)
+    perplexity_1 = lda.perplexity(X, distr, sub_sampling=False)
+
+    score = lda.score(X)
+    perplexity_2 = np.exp(-1. * (score / np.sum(X.data)))
+    assert_almost_equal(perplexity_1, perplexity_2)
+
+
+if __name__ == '__main__':
```

Comment:
+1, to run the tests on a specific file you can use nose directly:

```
nosetests -s -v sklearn/decomposition/tests/test_online_lda.py
```

and for a specific test:

```
nosetests -s -v sklearn/decomposition/tests/test_online_lda.py:test_lda_score_perplexity
```